### PR TITLE
fix(metadata): add hardcoded subtitle override

### LINKED ISSUE #165: Language handling
- [ ] Manual input when no language detected.
- [ ] missing `-hc` arg
- [ ] blu-ray discs are not being handled

### PR DESCRIPTION
**Pull request by Audionut**
## Summary

- restore -hc / --hardcoded-subs as a presence-aware manual metadata override
- centralize automatic hardcoded-subtitle detection into the finalized prepared fact used by tracker rules and PTP
- add the Web UI toggle, generated API contracts, regression tests, and public documentation

## Testing

- make test-go
- make precommit
- make e2e-build
- pnpm --dir documentation run check
- make backend

Closes #165

## Summary by CodeRabbit

* **New Features**
  * Added automatic detection of hardcoded subtitles from release and media metadata.
  * Added CLI support with `--hardcoded-subs` and `-hc` options.
  * Added a Web UI toggle to enable or disable hardcoded-subtitle handling.
  * Propagated hardcoded-subtitle status through uploads, validation, tracker rules, and release workflows.
* **Documentation**
  * Documented CLI and Web UI configuration, including tracker-specific requirements.
* **Bug Fixes**
  * Ensured explicit overrides consistently take precedence over automatic detection.

### CODE REVIEW

**Comment by Audionut** [comment]
@coderabbitai review

File: internal/metadata/media_details.go [hunk 1]
```diff
@@ -1638,6 +1645,23 @@ func containsExactHybrid(values []string) bool {
...
 }
 
+// hardcodedSubsFromMeta centralizes automatic marker detection before explicit metadata overrides.
+func hardcodedSubsFromMeta(meta preparationstate.State) bool {
+	values := make([]string, 0, 5+len(meta.FileList)+len(meta.Release.Edition)+len(meta.Release.Other))
+	values = append(values,
+		pathutil.Base(meta.SourcePath),
+		pathutil.Base(meta.VideoPath),
+		meta.Filename,
+		meta.ReleaseName,
```
[the remark this answers is not in the document]
Comment (Audionut) [reply internal/metadata/media_details.go:1655]: Fixed in 8b3e6e27d: local SourcePath and VideoPath basenames now use filepath.Base. Focused Go race tests and make lint/pathpolicy pass.

File: internal/trackers/impl/standalone/ptp/definition_test.go [hunk 2]
```diff
@@ -140,12 +141,13 @@ func TestPTPFreshUploadTaxonomy(t *testing.T) {
...
 	meta.ReleaseName = "Example.Release.2026.1080p.WEB-DL.x265-GRP"
 	meta.FileList = nil
+	meta.HardcodedSubs = false
 	meta.AudioLanguages = []string{"Japanese"}
 	meta.SubtitleLanguages = []string{"French"}
 	if got := resolveTrumpable(meta); len(got) != 1 || got[0] != 14 {
 		t.Fatalf("no-English trumpable=%#v", got)
 	}
-	meta.ReleaseName = "Example.Release.2026.1080p.WEB-DL.x265.HARDSUB-GRP"
+	meta.HardcodedSubs = true
```
[the remark this answers is not in the document]
Comment (Audionut) [reply internal/trackers/impl/standalone/ptp/definition_test.go:150]: Fixed in 8b3e6e27d: the explicit false override test now retains a HARDSUB marker and still asserts only trumpable value 14. Focused PTP race tests pass.

File: cmd/upbrr/cli_options_test.go [hunk 3]
```diff
@@ -973,6 +977,18 @@ func TestBuildCLIRequestMetadataOverrides(t *testing.T) {
 	if req.MetadataOverrides.Anime == nil || *req.MetadataOverrides.Anime {
 		t.Fatalf("expected not-anime to force anime=false, got %#v", req.MetadataOverrides.Anime)
 	}
+
+	falseOpts, falseVisited, falsePaths, err := parseCLIOptions([]string{"--hardcoded-subs=false", "movie.mkv"})
```
Comment (Audionut) [inline cmd/upbrr/cli_options_test.go:981]: @coderabbitai Addressed "Add a false-override CLI regression test" in 8b3e6e27d: --hardcoded-subs=false now asserts a non-nil false override. Focused CLI race tests pass.

File: pkg/api/prepared_release.go [hunk 4]
```diff
@@ -167,6 +167,8 @@ type MediaFacts struct {
 	Audio             string
 	Channels          string
 	Commentary        bool
+	// HardcodedSubs is resolved from automatic detection and then any explicit override.
```
Comment (Audionut) [inline pkg/api/prepared_release.go:170]: @coderabbitai Addressed "Document HardcodedSubs above the field" in 8b3e6e27d: the finalized detection-and-override contract now directly documents MediaFacts.HardcodedSubs. Precommit passes.

File: webui/src/pages/input/index.test.tsx [hunk 5]
```diff
@@ -304,5 +302,20 @@ describe("InputPage", () => {
...
+      ...facet,
+      view: {
+        ...facet.view,
+        intent: { ...facet.view.intent, metadata: { HardcodedSubs: true } },
+      },
+      changeMetadata: vi.fn(),
+    };
+    rerender(<InputPage facet={enabledFacet} {...pageProps} />);
+    fireEvent.click(screen.getByLabelText("Hardcoded subtitles"));
+    expect(enabledFacet.changeMetadata).toHaveBeenLastCalledWith({});
```
Comment (Audionut) [inline webui/src/pages/input/index.test.tsx:319]: @coderabbitai Addressed "Cover clearing the override" in 8b3e6e27d: the enabled-switch case now verifies that disabling removes HardcodedSubs. All 221 frontend unit tests and typecheck pass.

**Comment by Audionut** [comment]
from discord

Upbrr tags all releases uploaded to PTP without English audio and subs as No English Subtitles. There doesn't seem to be a cli-argument or at least a prompt to prevent that and you can't edit the tag after the upload in case of hardcoded English subs. Staff doesn't like it.
It would be good if we could explicitly tell Upbrr whether it shall mark an upload as No English Subtitles or Hardcoded Subtitles. An automatic With Commentary detection would be nice as well (if it doesn't exist yet).

**Comment by Audionut** [comment]
superseded by https://github.com/autobrr/upbrr/pull/450
